Swiftx/Combinators.swift

```diff
@@ -88,10 +88,16 @@ public func |> <A, B>(a : A, f : A -> B) -> B {
 /// point at which further application of x to a function is the same x.
 ///
 ///     x = f(x)
-public func fix<A>(f : ((A -> A) -> A -> A)) -> A -> A {
+public func fix<A, B>(f : (A -> B) -> A -> B) -> A -> B {
 	return { x in f(fix(f))(x) }
 }
 
+/// The fixpoint (or Y) combinator computes the least fixed point of an equation. That is, the first
+/// point at which further application of x to a function is the same x.
+/// fixt is the exception enabled version of fix.
+public func fixt<A, B>(f : (A throws -> B) -> A throws -> B) -> A throws -> B {
```

Comment:
Hi Robert,

I tried a straight overload, and it does work, but it seems to enforce
explicit type annotations of all `fix`'d functions.

I think it could break client code (requiring them to have to sprinkle
annotations throughout). Without explicit type annotations, I got
`ambiguous function call` errors.

On Sun, Sep 6, 2015 at 9:09 PM Robert Widmann <notifications@github.com>
wrote:

> In Swiftx/Combinators.swift
> <https://github.com/typelift/Swiftx/pull/30#discussion_r38825533>:
>
> >  	return { x in f(fix(f))(x) }
> >  }
> >
> > +/// The fixpoint (or Y) combinator computes the least fixed point of an equation. That is, the first
> > +/// point at which further application of x to a function is the same x.
> > +/// fixt is the exception enabled version of fix.
> > +public func fixt<A, B>(f : (A throws -> B) -> A throws -> B) -> A throws -> B {
>
> I wonder if a straight overload might be more prudent here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/typelift/Swiftx/pull/30/files#r38825533>.
>